storage/src/journal/benches/fixed_read.rs

```diff
@@ -0,0 +1,123 @@
+use commonware_runtime::tokio::{Blob, Config as TConfig, Context, Executor};
+use commonware_storage::journal::fixed::{Config as JConfig, Journal};
+use commonware_utils::array::FixedBytes;
+use criterion::{black_box, criterion_group, BatchSize, Criterion};
+use futures::future::try_join_all;
+use rand::{rngs::StdRng, Rng, RngCore, SeedableRng};
+use tokio::runtime::Runtime;
+
+/// Partition name to use in the journal config.
+const PARTITION: &str = "test_partition";
+
+/// Value of items_per_blob to use in the journal config.
+const ITEMS_PER_BLOB: u64 = 10000;
+
+/// Size of each journal item in bytes.
+const ITEM_SIZE: usize = 32;
+
+/// Number of items to write to the journal during setup before invoking the benchmark.
+const ITEMS_TO_WRITE: u64 = 10000000;
+
+/// Setup the benchmark by writing ITEMS_TO_WRITE random items to a journal of items with ITEM_SIZE
+/// bytes each. The journal is configured to use ITEMS_PER_BLOB items per blob.
+async fn bench_init(context: Context) -> Journal<Blob, Context, FixedBytes<ITEM_SIZE>> {
+    let journal_config = JConfig {
+        partition: PARTITION.to_string(),
+        items_per_blob: ITEMS_PER_BLOB,
+    };
+
+    // Initialize the journal.
+    let mut journal = Journal::init(context, journal_config).await.unwrap();
+    let sz = journal.size().await.unwrap();
+    assert_eq!(sz, 0);
+
+    // Append ITEMS_TO_WRITE random items to the journal
+    let mut rng = StdRng::seed_from_u64(0);
+    let mut arr = [0; ITEM_SIZE];
+    for _ in 0..ITEMS_TO_WRITE {
+        rng.fill_bytes(&mut arr);
+        journal
+            .append(FixedBytes::new(arr))
+            .await
+            .expect("failed to append data");
+    }
+    journal.sync().await.unwrap();
+
+    journal
+}
+
+async fn bench_setup(context: Context) -> Journal<Blob, Context, FixedBytes<ITEM_SIZE>> {
+    let journal_config = JConfig {
+        partition: PARTITION.to_string(),
+        items_per_blob: ITEMS_PER_BLOB,
+    };
+
+    Journal::init(context, journal_config).await.unwrap()
+}
+
+async fn bench_run_serial(
+    journal: &Journal<Blob, Context, FixedBytes<ITEM_SIZE>>,
+    items_to_read: usize,
+) {
+    let mut rng = StdRng::seed_from_u64(0);
+
+    for _ in 0..items_to_read {
+        let pos = rng.gen_range(0..ITEMS_TO_WRITE);
+        black_box(journal.read(pos).await.expect("failed to read data"));
+    }
+}
+
+async fn bench_run_parallel(
+    journal: &Journal<Blob, Context, FixedBytes<ITEM_SIZE>>,
+    items_to_read: usize,
+) {
+    let mut rng = StdRng::seed_from_u64(0);
+
+    let mut futures = Vec::new();
+    for _ in 0..items_to_read {
+        let pos = rng.gen_range(0..ITEMS_TO_WRITE);
+        futures.push(journal.read(pos));
+    }
+    try_join_all(futures).await.expect("failed to read data");
+}
+
+fn bench_fixed_read(c: &mut Criterion) {
+    let runtime_cfg = TConfig::default();
+    let (_, context) = Executor::init(runtime_cfg.clone());
+    let runtime = Runtime::new().unwrap();
```

Comment:
It feels weird to me that you instantiate 2 tokio runtimes 🤔

Reply:
I think it may be cleaner to call `executor.start()` with the commonware-runtime in the `b` below

Reply:
Should also enable you to remove `features = ["async_tokio"]`

Reply:
This test involves a lot of async IO during setup as well as during the benchmark itself. I am not sure how to share the commonware-runtime across both without instantiating it here, and unfortunately executor.start() can't take a commonware-runtime.

Reply:
I was able to get rid of the separate tokio runtime by making our tokio commonware-runtime implement criterion::AsyncExecutor, allowing me to pass it to to_async. I think you had some concerns with doing this though?